packages/number-field/src/vaadin-number-field.js

```diff
@@ -467,9 +492,14 @@ export class NumberField extends InputFieldMixin(SlotStylesMixin(ThemableMixin(E
   checkValidity() {
     if (
       this.inputElement &&
-      (this.required || this.min !== undefined || this.max !== undefined || this.__validateByStep)
+      (this.required || this.min !== undefined || this.max !== undefined || this.step !== undefined)
     ) {
-      return this.inputElement.checkValidity();
+      if (this.value == null || this.value == '') {
+        return super.checkValidity();
+      } else {
+        // Mimic validation logic provided by native `<input type="number">` as we don't use it.
+        return !(this.value > this.max || this.value < this.min || !this.__isValidByStep(this.value));
```

Comment:
This does not cover the native input validation logic regarding localization. Setting a min of `1.5` and entering a value of `1,3`, which is an internal value of `1.3`, does not mark the input invalid.

Also `value` is still a string, while `min` and `max` are numbers, which indicates an implicit type coersion in the comparison. Value should be parsed to a number instead, which of course requires us to take the users locale into account.

I also could not find any tests regarding this validation. I assume the previous version relied on the fact that the browser implemented this check. Now that we are doing this on our own we would need tests to cover it.